### PR TITLE
Fixes to `snow ws migrate`

### PR DESCRIPTION
**Pull request by sfc-gh-jsikorski**
### Pre-review checklist
   * [x] I've confirmed that instructions included in README.md are still correct after my changes in the codebase.
   * [x] I've added or updated automated unit tests to verify correctness of my new code.
   * [ ] I've added or updated integration tests to verify correctness of my new code.
   * [x] I've confirmed that my changes are working by executing CLI's commands manually on MacOS.
   * [ ] I've confirmed that my changes are working by executing CLI's commands manually on Windows.
   * [x] I've confirmed that my changes are up-to-date with the target branch.
   * [ ] I've described my changes in the release notes.
   * [x] I've described my changes in the section below.

### Changes description
As part of post bug-bash fixes, this PR aims to allow user to migrate V1 `snowflake.yml` files, that contain both snowpark and streamlit definitions.
Part of the work, was to move all conversion logic into the `api` module, to simplify it and limit code duplication

### CODE REVIEW

File: tests/workspace/test_manager.py [hunk 1]
```diff
@@ -86,7 +86,7 @@ def test_bundle_of_invalid_entity_type(temp_dir):
 
 @pytest.mark.parametrize(
     "project_directory_name",
-    ["migration_streamlit_V1_to_V2", "migration_snowpark_V1_to_V2"],
+    ["migration_streamlit_v1_to_v2", "migration_snowpark_V1_to_V2"],
```
Comment (sfc-gh-astus) [inline tests/workspace/test_manager.py:89]: Why changed only in streamlit?

Comment (sfc-gh-jsikorski) [reply tests/workspace/test_manager.py:89]: ruff was having problems with this name (don't ask me why) - https://github.com/snowflakedb/snowflake-cli/actions/runs/10618285380/job/29433633082

Comment (sfc-gh-astus) [reply tests/workspace/test_manager.py:89]: So could you change V->v in snowpark too?

Comment (sfc-gh-jsikorski) [reply tests/workspace/test_manager.py]: Done

**Review by sfc-gh-astus — APPROVED** [review]

**Pull request merged**